### PR TITLE
Mode base reporting change in supported modes

### PR DESCRIPTION
**Pull request by hicklin**
Adds a Mode Base cluster instance method that allows the SDK consumer to easily report that the contents of the supported modes attribute has changed.

Adds documentation to the relevant virtual functions that the SDK consumer implements to remind them that they should call this reporting function if the contents of the supported modes attribute change.

**NOTE**: This depends on PR #28573 to be merged.

### CODE REVIEW

File: src/app/clusters/mode-base-server/mode-base-server.cpp [hunk 1]
```diff
@@ -23,7 +23,6 @@
 #include <app/clusters/on-off-server/on-off-server.h>
 #include <app/reporting/reporting.h>
 #include <app/util/attribute-storage.h>
-#include <app/util/config.h>
```
Comment (bzbarsky-apple) [inline src/app/clusters/mode-base-server/mode-base-server.cpp:26]: Why is config.h being removed?  That's what pulls in the definition of `EMBER_AF_PLUGIN_ON_OFF_SERVER`.....

File: src/app/clusters/mode-base-server/mode-base-server.h [hunk 2]
```diff
@@ -100,6 +102,13 @@ class Instance : public CommandHandlerInterface, public AttributeAccessInterface
     static constexpr uint8_t kMaxModeLabelSize = 64;
     static constexpr uint8_t kMaxNumOfModeTags = 8;
 
+    // List change reporting
+    /**
+     * Reports that the contents of the supported modes attribute has changed.
```
Comment (bzbarsky-apple) [inline src/app/clusters/mode-base-server/mode-base-server.h:107]: "have changed", if it's "contents".  Or "content" and then "has changed"...

File: src/app/clusters/mode-base-server/mode-base-server.h [hunk 3]
```diff
@@ -194,6 +203,9 @@ class Delegate
      * CopyCharSpanToMutableCharSpan to copy into the MutableCharSpan.
      * @return Returns a CHIP_NO_ERROR if there was no error and the label was returned successfully.
      * CHIP_ERROR_PROVIDER_LIST_EXHAUSTED if the modeIndex in beyond the list of available labels.
+     *
+     * Note: This is used by the SDK to populate the supported modes attribute. If the contents of this list changes,
```
Comment (bzbarsky-apple) [inline src/app/clusters/mode-base-server/mode-base-server.h:207]: Again, number agreement please, between "contents" and "changes".

**Review by bzbarsky-apple — APPROVED** [review]

File: src/app/clusters/mode-base-server/mode-base-server.h (hunk 3 above)
Comment (hicklin) [reply src/app/clusters/mode-base-server/mode-base-server.h]: Fixed.

**Review by andy31415 — APPROVED** [review]

**Pull request merged**